bin/improver-regrid

```diff
@@ -0,0 +1,116 @@
+                                         'performed using iris.analysis.'
+                                         'Nearest() instead of Linear(). '
+                                         'Use for less continuous fields, e.g.'
+                                         ' precipitation.')}),
+                              (['--extrapolation_mode'],
+                               {'default': 'nanmask',
+                                'help': ('Mode to use for extrapolating data '
+                                         'into regions beyond the limits of '
+                                         'the source_data domain. Modes are: '
+                                         'extrapolate - The extrapolation '
+                                         'points will take their value from '
+                                         'the nearest source point. '
+                                         'nan - The extrapolation points will '
+                                         'be be set to NaN. error - A '
+                                         'ValueError exception will be raised,'
+                                         ' notifying an attempt to extrapolate'
+                                         '. mask  - The extrapolation points '
+                                         'will always be masked, even if the '
+                                         'source data is not a MaskedArray. '
+                                         'nanmask - If the source data is a '
+                                         'MaskedArray the extrapolation points'
+                                         ' will be masked. Otherwise they will'
+                                         ' be set to NaN. Defaults to nanmask.'
+                                         )}),
+                              ]
+
+    cli_definition = {'central_arguments': [],
+                      'specific_arguments': cli_specific_arguments,
+                      'description': ('Regrid data from source_data on to the '
+                                      'grid contained within target_grid using'
+                                      ' iris.analysis.Linear() or optionally '
+                                      'iris.analysis.Nearest()')}
+
+    args = ArgParser(**cli_definition).parse_args()
+
+    source_data = load_cube(args.source_data_filepath)
+    target_grid = load_cube(args.target_grid_filepath)
+
+    regridder = iris.analysis.Linear(
+        extrapolation_mode=args.extrapolation_mode)
+    if args.nearest:
+        regridder = iris.analysis.Nearest(
+            extrapolation_mode=args.extrapolation_mode)
+
+    source_on_target = source_data.regrid(target_grid, regridder)
+    target_grid_attributes = {k: v for (k, v) in target_grid.attributes.items()
+                              if 'mosg__grid' in k}
+    amend_metadata(source_on_target, revised_attributes=target_grid_attributes)
+    delete_attributes(source_on_target, ['title'])
```

Comment:
Not using the new pattern match delete attributes functionality any more? If so should we downgrade what it does?

Reply:
Pattern matching or not, this needs to take a list to enable multiple attributes to be deleted with a single call. The pattern matching itself is not adding any code. The if statement that checks for a partial match is still required to check that a complete attribute matches to avoid the raising of a KeyError. If I remove ```if pattern in k``` I'll need to add a try/except KeyError wrapper around the code to deal with attempts to remove unset attributes. Can we leave it alone?